custom/enikshay/management/commands/base_data_dump.py

```diff
@@ -117,24 +122,29 @@ def generate_dump(self):
     def save_dump_to_blob(self, temp_path):
         with open(temp_path, 'rb') as file_:
             blob_db = get_blob_db()
-            blob_db.put(file_, self.result_file_name, timeout=60 * 48)  # 48 hours
-
-            file_format = Format.from_format(Format.CSV)
-            blob_dl_object = expose_blob_download(
+            blob_db.put(
+                file_,
                 self.result_file_name,
-                mimetype=file_format.mimetype,
-                content_disposition=safe_filename_header(self.result_file_name, file_format.extension),
-            )
+                timeout=60 * 48)  # 48 hours
+
+        file_format = Format.from_format(Format.CSV)
+        file_name_header = safe_filename_header(
+            self.result_file_name, file_format.extension)
+        blob_dl_object = expose_blob_download(
+            self.result_file_name,
+            mimetype=file_format.mimetype,
+            content_disposition=file_name_header
+        )
         return blob_dl_object.download_id
 
     def email_result(self, download_id):
-        if not self.recipient:
-            return
-        url = reverse('ajax_job_poll', kwargs={'download_id': download_id})
+        url = "%s%s?%s" % (get_url_base(),
+                           reverse('retrieve_download', kwargs={'download_id': download_id}),
+                           "get_file")  # downloads immediately, rather than rendering page
```

Comment:
FYI we also have a util that wraps this sort of thing: https://github.com/dimagi/commcare-hq/blob/master/corehq/util/view_utils.py#L116-L116

Reply:
Ooh, guess I imported the wrong reverse. I was confused as to why that didn't just work